mllib/src/main/scala/org/apache/spark/ml/classification/MultinomialLogisticRegression.scala

```diff
@@ -0,0 +1,611 @@
+          val msg = s"${optimizer.getClass.getName} failed."
+          logError(msg)
+          throw new SparkException(msg)
+        }
+        bcFeaturesStd.destroy(blocking = false)
+
+        /*
+           The coefficients are trained in the scaled space; we're converting them back to
+           the original space.
+           Note that the intercept in scaled space and original space is the same;
+           as a result, no scaling is needed.
+         */
+        var interceptSum = 0.0
+        var coefSum = 0.0
+        val rawCoefficients = Vectors.fromBreeze(state.x)
+        val (coefMatrix, interceptVector) = rawCoefficients match {
+          case dv: DenseVector =>
+            val coefArray = Array.tabulate(numClasses * numFeatures) { i =>
+              val flatIndex = if ($(fitIntercept)) i + i / numFeatures else i
+              val featureIndex = i % numFeatures
+              val unscaledCoef = if (featuresStd(featureIndex) != 0.0) {
+                dv(flatIndex) / featuresStd(featureIndex)
+              } else {
+                0.0
+              }
+              coefSum += unscaledCoef
+              unscaledCoef
+            }
+            val interceptVector = if ($(fitIntercept)) {
+              Vectors.dense(Array.tabulate(numClasses) { i =>
+                val coefIndex = (i + 1) * numFeaturesPlusIntercept - 1
+                val intercept = dv(coefIndex)
+                interceptSum += intercept
+                intercept
+              })
+            } else {
+              Vectors.sparse(numClasses, Seq())
+            }
+            (new DenseMatrix(numClasses, numFeatures, coefArray, isTransposed = true),
+              interceptVector)
+          case sv: SparseVector =>
+            throw new IllegalArgumentException("SparseVector is not supported for coefficients")
+        }
+
+        /*
+          When no regularization is applied, the coefficients lack identifiability because
+          we do not use a pivot class. We can add any constant value to the coefficients and
+          get the same likelihood. So here, we choose the mean centered coefficients for
+          reproducibility. This method follows the approach in glmnet, described here:
+
+          Friedman, et al. "Regularization Paths for Generalized Linear Models via
+            Coordinate Descent," https://core.ac.uk/download/files/153/6287975.pdf
+         */
+        if ($(regParam) == 0.0) {
+          val coefficientMean = coefSum / (numClasses * numFeatures)
+          coefMatrix.update(_ - coefficientMean)
+        }
+        /*
+          The intercepts are never regularized, so we always center the mean.
+         */
+        val interceptMean = interceptSum / numClasses
+        interceptVector match {
+          case dv: DenseVector => (0 until dv.size).foreach { i => dv.toArray(i) -= interceptMean }
+          case sv: SparseVector =>
+            (0 until sv.numNonzeros).foreach { i => sv.values(i) -= interceptMean }
```

Comment:
This can not be a sparse vector once the `interceptVector` is centered. 

```scala
val interceptVector = if (intercepts.length > 1) {
  val interceptMean = intercepts.sum / intercepts.length
  for (i = 0 until i < intercepts.length) {
    intercepts(i) -= interceptMean
  }
} else {
  Vectors.sparse(numClasses, Seq())
}
```